Translate docs/event-handling.md (#19)
reactjs/nl.reactjs.org: content/docs/handling-events.md

---
id: handling-events
title: Omgaan met Events
permalink: docs/handling-events.html
prev: state-and-lifecycle.html
next: conditional-rendering.html
redirect_from:
  - "docs/events-ko-KR.html"
---

Omgaan met events met React-elementen lijkt sterk op omgaan met events met DOM-elementen.

> Er zijn enkele syntactische verschillen:

Reviewer: ```suggestion
Maar, er zijn enkele syntactische verschillen:
```

Or add the word "Echter". We're making a 'tegenstelling' in this sentence compared to the previous, so it's proper to make that clear in the syntax as well.

Author: True, although the English version should technically have a "but" or "however" as well.

Reviewer: I agree, but the English language is more forgiving in this, while Dutch is less so. So I think it's better to make it explicit, especially if we're also targetting less experiences devs.

> * React events worden in camelCase benoemd in plaats van kleine letters.

Reviewer: ```suggestion
* React events worden in camelCase geschreven in plaats van kleine letters.
```

Author: I would make it clear that it's about the name of React events:
```suggestion
* De naam van React events wordt in camelCase geschreven in plaats van kleine letters.
```
or do you think your version implies this enough?

Reviewer: Yours is better. I lost the implicit meaning of "naming" of the original content. Nice one.

* Met JSX geef je een functie door als de event handler in plaats van een string.

> Bijvoorbeeld de HTML:

Reviewer: ```suggestion
Als voorbeeld, de HTML:
```

```html
<button onclick="activateLasers()">
  Activate Lasers
</button>
```

> is licht verschillend in React:

Reviewer: ```suggestion
verschilt lichtelijk in React:
```

}
```

Hier is `e` een synthetisch event.
React definieert deze synthetische events volgens de [W3C-specificatie (Engels)](https://www.w3.org/TR/DOM-Level-3-Events/), zodat je je geen zorgen hoeft te maken over cross-browser compatibiliteit.
Zie de referentiegids [`SyntheticEvent`](/docs/events.html) voor meer informatie.

Wanneer je React gebruikt, zou je over het algemeen `addEventListener` niet moeten aanroepen om listeners toe te voegen aan een DOM-element nadat het is aangemaakt.
Voorzie in plaats daarvan alleen een listener wanneer het element voor het eerst wordt gerenderd.

> Wanneer je een component definieert met een [ES6-klasse](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Classes), is een algemeen patroon dat een event handler een methode van de klasse is. Dit `Toggle`-component rendert bijvoorbeeld een knop waarmee de gebruiker kan wisselen tussen "ON"- en "OFF"-toestanden:

Reviewer: new line

);
```

[**Probeer het op CodePen**](http://codepen.io/gaearon/pen/xEmzGg?editors=0010)

> Je bent best voorzichtig met de betekenis van `this` in JSX callbacks.

Reviewer: ❗️ 
```suggestion
Je moet voorzichtig zijn met de betekenis van `this` in JSX callbacks.
```

Author: Well, I thought I'd be gentle 🤷‍♂️

Reviewer: It's not necessarily about being harsh or gentle. The original sentence is a warning to the user regarding the meaning of `this` in callbacks. That alerting meaning is lost in your translation.

> In JavaScript zijn klassenmethoden standaard niet [bound (Engels)](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Function/bind).

Reviewer: `bound` or `gebonden`?

}
```

> Het probleem met deze syntax is dat elke keer dat de `LoggingButton` wordt gerenderd een andere callback wordt aangemaakt.

Reviewer: ```suggestion
Het probleem van deze syntax is dat elke keer dat de `LoggingButton` wordt gerenderd een andere callback wordt aangemaakt.
```

In de meeste gevallen is dit prima.

> Echter, als deze callback wordt doorgegeven als prop naar lagere componenten, is het mogelijk dat deze componenten een extra re-render doen.

Reviewer: ```suggestion
Echter, als deze callback wordt doorgegeven als prop naar lagere componenten, is het mogelijk dat deze componenten een extra re-render uitvoeren.
```

> Over het algemeen raden we aan te binden in de constructor of door de class fields syntax te gebruiken om dit soort performance problemen te voorkomen.

Reviewer: ```suggestion
Over het algemeen raden we aan te binden in de constructor of de class fields syntax te gebruiken om dit soort performance problemen te voorkomen.
```

## Argumenten Doorgeven aan Event Handlers {#passing-arguments-to-event-handlers}

Binnen een loop is het gebruikelijk om een extra parameter te willen doorgeven aan een event handler.
Als `id` bijvoorbeeld de rij-ID is, zou elk van de volgende werken:

```js
<button onClick={(e) => this.deleteRow(id, e)}>Delete Row</button>
<button onClick={this.deleteRow.bind(this, id)}>Delete Row</button>
```

De bovenstaande twee regels zijn equivalent en gebruiken respectievelijk [pijlfuncties](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Functions/Arrow_functions) en [`Function.prototype.bind`](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Function/bind).

> In beide gevallen wordt het `e`-argument dat de React event vertegenwoordigt, als een tweede argument doorgegeven na de ID.

Reviewer: ```suggestion
In beide gevallen wordt het `e`-argument dat de React event vertegenwoordigt als een tweede argument doorgegeven na de ID.
```
